cluster-autoscaler/utils/deletetaint/delete.go

```diff
@@ -31,27 +32,39 @@ import (
 const (
 	// ToBeDeletedTaint is a taint used to make the node unschedulable.
 	ToBeDeletedTaint = "ToBeDeletedByClusterAutoscaler"
+
+	maxConflictRetries    = 50
+	conflictRetryInterval = 250 * time.Millisecond
 )
 
 // MarkToBeDeleted sets a taint that makes the node unschedulable.
 func MarkToBeDeleted(node *apiv1.Node, client kube_client.Interface) error {
-	// Get the newest version of the node.
-	freshNode, err := client.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
-	if err != nil || freshNode == nil {
-		return fmt.Errorf("failed to get node %v: %v", node.Name, err)
-	}
+	try := 0
+	for {
+		// Get the newest version of the node.
+		freshNode, err := client.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
+		if err != nil || freshNode == nil {
+			return fmt.Errorf("failed to get node %v: %v", node.Name, err)
+		}
 
-	added, err := addToBeDeletedTaint(freshNode)
-	if added == false {
-		return err
-	}
-	_, err = client.CoreV1().Nodes().Update(freshNode)
-	if err != nil {
-		glog.Warningf("Error while adding taints on node %v: %v", node.Name, err)
-		return err
+		added, err := addToBeDeletedTaint(freshNode)
+		if added == false {
+			return err
+		}
+		_, err = client.CoreV1().Nodes().Update(freshNode)
+		if err != nil && errors.IsConflict(err) && try < maxConflictRetries {
```

Comment:
Updated to just retry for 1 minute instead of counting retries.

Reply:
I don't know if it's worth it making these settings configurable? In our experience 1-2 retries are usually enough, the deadline is only in case something goes completely wrong.

Reply:
Also 50 retries seem like much too much. Did you actually observe that you need that many retries to perform update?

Reply:
Disregard my last comment. YOu already answered that :)

Reply:
> In our experience 1-2 retries are usually enough, the deadline is only in case something goes completely wrong.

Given that I believe we want to have deadline at sth around 2 seconds not 60. 
If something is terribly wrong we want to exit with error. And not attack api server with request.

Reply:
Fixed, it's 5 seconds now (with a 750ms interval).